flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java

```diff
@@ -306,6 +307,29 @@ private static void assignTaskStatesToOperatorInstances(
 	}
 
 	/**
+	 * Determine the subset of {@link KeyGroupsStateHandle KeyGroupsStateHandles} with correct
+	 * key group index for the given subtask {@link KeyGroupRange}.
+	 * <p>
+	 * <p>This is publicly visible to be used in tests.
+	 */
+	public static List<KeyedStateHandle> getKeyedStateHandles(
```

Comment:
Could we completely remove `getKeyGroupsStateHandles()` and remove everything with this method that works on `KeyedStateHandle`? This includes using also only `KeyedStateHandle` for the raw keyed state that is checkpointed into streams (without backend). Otherwise, this is almost duplicated code. What do you think?